Add sequence diagrams for init, join, and reset

dlipovetsky opened the pull request:
Images generated by sequencediagram.org

astrieanna commented on docs/diagrams/join.md:
```diff
@@ -0,0 +1,41 @@
+end
+
+group create certificates
+etcdadm->etcdadm: create CA certificate
```
Join doesn't create a CA cert, right? It uses the CA cert passed in by the user, the one that was created in init?

dlipovetsky replied:
Right. Join doesn't create a CA cert. Thanks for catching this!

astrieanna commented on docs/diagrams/reset.md:
```diff
@@ -0,0 +1,32 @@
+title etcdadm join https://member1:2379
```
I think this label is wrong (should be reset)

dlipovetsky replied:
You're right. Copy-paste error. Thanks!

astrieanna commented on docs/diagrams/join.md:
```diff
@@ -0,0 +1,41 @@
+participant etcdadm
+participant systemd
+database member1
+database member2
```
Because the sequence diagram is showing the interaction between multiple processes on one host (etcdadm, systemd, member2) and also between hosts, it feels a bit confusing. It looks like the website you're using has a `participantgroup` concept. Do you think grouping the process participants by host would help?

dlipovetsky replied:
Yes! Great idea. Will add this.

astrieanna commented on docs/diagrams/init.md:
```diff
@@ -0,0 +1,32 @@
+end
+
+group install artifacts
+etcdadm->host1: copy etcd, etcdctl from local cache to install directory
```
Two of the participants in this diagram (`etcadm`, `systemd`) are processes; `host1` is not. In order to make the participants more parallel/similar, maybe `host1` here should be a `database` participant or be called the `filesystem of host1`.

dlipovetsky replied:
Good idea. I'll use `filesystem` in the `host1` `participantgroup`.

astrieanna commented on docs/diagrams/reset.md:
```diff
@@ -0,0 +1,32 @@
+end
+
+group destroy certificates
+etcdadm->etcdadm: destroy CA certificate
```
If `etcdadm reset` deletes the CA cert, then if you want to run `etcdadm join` again on the same machine (to the same cluster), you'll need to copy the CA cert over again, right?

dlipovetsky replied:
Yes, that's what currently happens.

dlipovetsky commented:
@astrieanna Thanks a lot for your review. I tried to address your comments.

justinsb commented:
Thanks @dlipovetsky - looks great.  One suggestion would be to include a note in the repo about how to generate the pngs

justinsb commented:
/approve
/lgtm

astrieanna commented:
It looks like in addressing the review comments, the image files were not updated.

dlipovetsky commented:
@astrieanna My fault. I updated the image files. I also changed to PlantUML so they can be re-generated at the command line.

@justinsb Good point. I added a make target to generate diagrams.

astrieanna commented:
These diagrams look great. :)
